src/app/public/modules/back-to-top/back-to-top-adapter.service.ts

```diff
@@ -19,23 +26,67 @@ import {
   takeUntil
 } from 'rxjs/operators';
 
+import {
+  SkyBackToTopComponent
+} from './back-to-top.component';
+
+import {
+  SkyBackToTopType
+} from './models/back-to-top-type';
+
 /**
  * @internal
  */
 @Injectable()
 export class SkyBackToTopDomAdapterService implements OnDestroy {
 
   private ngUnsubscribe = new Subject<void>();
+  private renderer: Renderer2;
 
   constructor(
-    private windowRef: SkyAppWindowRef
-  ) { }
+    private componentFactoryResolver: ComponentFactoryResolver,
+    private applicationRef: ApplicationRef,
+    private injector: Injector,
+    private windowRef: SkyAppWindowRef,
+    private rendererFactory: RendererFactory2
+  ) {
+    // Based on suggestions from https://github.com/angular/angular/issues/17824
+    // for accessing an instance of Renderer2 in a service since Renderer2 can't
+    // be injected into a service.  Passing undefined for both parameters results
+    // in the default renderer which is what we want here.
+    this.renderer = this.rendererFactory.createRenderer(undefined, undefined);
+  }
 
   public ngOnDestroy(): void {
     this.ngUnsubscribe.next();
     this.ngUnsubscribe.complete();
   }
 
+  public createComponentWithinModal(elementRef: ElementRef): ComponentRef<SkyBackToTopComponent> {
+    const componentRef = this.componentFactoryResolver
+    .resolveComponentFactory<SkyBackToTopComponent>(SkyBackToTopComponent)
+    .create(this.injector);
+
+    this.applicationRef.attachView(componentRef.hostView);
+
+    const modal = this.findModalElement(elementRef.nativeElement);
+    const modalFooter = modal.querySelector('.sky-modal-footer');
+    this.renderer.insertBefore(modal, this.getRootNode(componentRef), modalFooter);
+
+    return componentRef;
+  }
+
+  public destroyCompoennt(componentRef: ComponentRef<SkyBackToTopComponent>): void {
```

Comment:
`destroyCompoennt` typo

Reply:
Also, same comment applies here - do we usually handle destroying components in an adapter service? Or would this have a better fit elsewhere?

Reply:
See my comment above. Fixed the typo.